src/nbl/ext/DebugDraw/CDrawAABB.cpp

```diff
@@ -0,0 +1,374 @@
+	    uint32_t instancesByteOffset = imaginaryChunk.alloc_addr(sizeof(InstanceData) * instanceCount, sizeof(InstanceData));
+	    const uint32_t totalSize = imaginaryChunk.get_allocated_size();
+
+	    inputOffset = SCachedCreationParameters::streaming_buffer_t::invalid_value;
+		std::chrono::steady_clock::time_point waitTill = std::chrono::steady_clock::now() + std::chrono::milliseconds(1u);
+	    streaming->multi_allocate(waitTill, 1, &inputOffset, &totalSize, &MaxAlignment);
+
+	    memcpy(streamingPtr + vertexByteOffset, m_unitAABBVertices.data(), sizeof(m_unitAABBVertices[0]) * m_unitAABBVertices.size());
+	    memcpy(streamingPtr + instancesByteOffset, std::addressof(*instancesIt), sizeof(InstanceData) * instanceCount);
+		instancesIt += instanceCount;
+
+	    assert(!streaming->needsManualFlushOrInvalidate());
+
+	    SPushConstants pc;
+	    memcpy(pc.MVP, cameraMat3x4, sizeof(pc.MVP));
+	    pc.pVertexBuffer = m_cachedCreationParams.streamingBuffer->getBuffer()->getDeviceAddress() + vertexByteOffset;
+	    pc.pInstanceBuffer = m_cachedCreationParams.streamingBuffer->getBuffer()->getDeviceAddress() + instancesByteOffset;
+
+	    commandBuffer->pushConstants(m_pipeline->getLayout(), ESS_VERTEX, 0, sizeof(SPushConstants), &pc);
+	    commandBuffer->draw(m_unitAABBVertices.size(), instanceCount, 0, 0);
+
+	    streaming->multi_deallocate(1, &inputOffset, &totalSize, waitInfo);
+    }
+
+	return true;
+}
+
+std::array<float32_t3, 24> DrawAABB::getVerticesFromAABB(const core::aabbox3d<float>& aabb)
+{
+	const auto& pMin = aabb.MinEdge;
+	const auto& pMax = aabb.MaxEdge;
+
+	std::array<float32_t3, 24> vertices;
+	vertices[0] = float32_t3(pMin.X, pMin.Y, pMin.Z);
+	vertices[1] = float32_t3(pMax.X, pMin.Y, pMin.Z);
+	vertices[2] = float32_t3(pMin.X, pMin.Y, pMin.Z);
+	vertices[3] = float32_t3(pMin.X, pMin.Y, pMax.Z);
+
+	vertices[4] = float32_t3(pMax.X, pMin.Y, pMax.Z);
+	vertices[5] = float32_t3(pMax.X, pMin.Y, pMin.Z);
+	vertices[6] = float32_t3(pMax.X, pMin.Y, pMax.Z);
+	vertices[7] = float32_t3(pMin.X, pMin.Y, pMax.Z);
+
+	vertices[8] = float32_t3(pMin.X, pMax.Y, pMin.Z);
+	vertices[9] = float32_t3(pMax.X, pMax.Y, pMin.Z);
+	vertices[10] = float32_t3(pMin.X, pMax.Y, pMin.Z);
+	vertices[11] = float32_t3(pMin.X, pMax.Y, pMax.Z);
+
+	vertices[12] = float32_t3(pMax.X, pMax.Y, pMax.Z);
+	vertices[13] = float32_t3(pMax.X, pMax.Y, pMin.Z);
+	vertices[14] = float32_t3(pMax.X, pMax.Y, pMax.Z);
+	vertices[15] = float32_t3(pMin.X, pMax.Y, pMax.Z);
+
+	vertices[16] = float32_t3(pMin.X, pMin.Y, pMin.Z);
+	vertices[17] = float32_t3(pMin.X, pMax.Y, pMin.Z);
+	vertices[18] = float32_t3(pMax.X, pMin.Y, pMin.Z);
+	vertices[19] = float32_t3(pMax.X, pMax.Y, pMin.Z);
+
+	vertices[20] = float32_t3(pMin.X, pMin.Y, pMax.Z);
+	vertices[21] = float32_t3(pMin.X, pMax.Y, pMax.Z);
+	vertices[22] = float32_t3(pMax.X, pMin.Y, pMax.Z);
+	vertices[23] = float32_t3(pMax.X, pMax.Y, pMax.Z);
+
+	return vertices;
+}
```

Comment:
function not needed, you shall store this array in vertex shader and index it with `gl_VertexIndex`